Backlight.cpp

```diff
@@ -113,7 +113,11 @@ void showBacklight(){
       break;
     case 255: //loop
       fadeToBlackBy(backlight_leds, BACKLIGHT_NUM_LEDS , 255 - backlightBrightness );
-      backlight_leds[ledIndex[loopIndex++]] = backlight1;
+      if (loopIndex >= BACKLIGHT_NUM_LEDS) {
+        loopIndex = 0; // Reset loop index when it reaches the end
+      }
+      backlight_leds[ledIndex[loopIndex]] = backlight1;
+      loopIndex++;
```

Comment:
```suggestion
      loopIndex = (loopIndex + 1) % BACKLIGHT_NUM_LEDS;
      backlight_leds[ledIndex[loopIndex]] = backlight1;
```
this should work as well and shouldnt need to add more lines of code. The entire codebase is a mess though but I cant really be bothered to refactor it